src/ResourceManager/Resources/Commands.Resources/RoleAssignments/ValidateScope.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Linq;
+using System.Management.Automation;
+
+namespace Microsoft.WindowsAzure.Commands.Common
+{
+    [AttributeUsage(AttributeTargets.Property | AttributeTargets.Field)]
+    public class ValidateScope : ValidateArgumentsAttribute
+    {
+        protected override void Validate(object arguments, EngineIntrinsics engineIntrinsics)
+        {
+            if (arguments != null)
+            {
+                string scope = (string)arguments;
+
+                if (scope.Length == 0 || !scope.StartsWith("/"))
+                {
+                    throw new ValidationMetadataException("Scope must begin with '/'.");
+                }
+
+                // "/" is a valid scope.
+                if (string.Compare(scope, "/") == 0)
+                {
+                    return;
+                }
+
+                // For scope that on tenant level, we don't perform further check here.
+                if (!scope.StartsWith("/subscriptions", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    return;
+                }
+
+                var parts = scope.Substring(1).Split('/');   // Skip the leading '/'
+
+                if (parts.Contains(string.Empty))
+                {
+                    throw new ValidationMetadataException("Name or value in Scope must be not empty.");
+                }
+
+                int count = parts.Count();
+
+                if (count % 2 != 0)
+                {
+                    throw new ValidationMetadataException("Scope should have even number of parts.");
+                }
+
+                Guid subscriptionId;
+                if (!Guid.TryParseExact(parts[1], "D", out subscriptionId)  //32 digits separated by hyphens: 00000000-0000-0000-0000-000000000000
+                    || subscriptionId == Guid.Empty)                    
+                {
+                    throw new ValidationMetadataException("Invalid or empty subscriptionId.");
+                }
+
+                if (count >= 4 && string.Compare(parts[2], "resourceGroups", true) != 0)
+                {
+                    throw new ValidationMetadataException("Scope should begin with \"/subscriptions/<subscription id>/resourceGroups/\".");
```

Comment:
need to figure out the call for
New-AzureRmRoleAssignment -ObjectId $id -RoleDefinitionId "8e3af657-a8ff-443c-a75c-2fe8c4bcb635" -Scope "/subscriptions/4004a9fd-d58e-48dc-aeb2-4a4aec58606f/providers/Microsoft.Authorization"

---
In reply to: [118789162](https://github.com/Azure/azure-powershell/pull/4019#discussion_r118789162) [](ancestors = 118789162)